### PR TITLE
Add color dropdown to roles modal

### LINKED ISSUE #424: My Roles: Dropdown
**As a** peoplemover user
**I would** like a neater experience when creating roles
**so that** I am not confused by several line breaks (currently)

## Acceptance Criteria

**Given** I am in a space
**When** I click My Roles
**Then** I see the new My Roles Modal
**And** when I click on edit icon, two boxes appear around the role name and color
**And** when I click on color circle, I see a dropdown to change color
**And** on the dropdown, I can see what's being selected with white and grey outline

## Accessibility
Create aria labels with color name

## Design
https://www.figma.com/file/S4XKeovlQqeRk64k2qzhf8gL/PeopleMover?node-id=5269%3A611

### PR DESCRIPTION
**Pull request by RandomAshes**
## Issue
Resolves #424

## What was done
- [x] Add color dropdown to roles modal

## How to test
1. Ensure that the color dropdown works properly on the roles modal
2. Ensure that the roles and tags modals work as expected

Feature URL: https://featurebranchui.apps.pd01i.edc1.cf.ford.com/

### CODE REVIEW

**Review by michaelrygiel — APPROVED** [review]
Need to make a ticket to work on accessibility for dropdown select. Other than that, its 👌

File: ui/src/Traits/EditTraitSection.tsx [hunk 1]
```diff
@@ -196,24 +198,6 @@ function EditTraitSection({
...
-                        colorRefs.push(ref);
-
-                        return (
-                            <span key={index}
-                                ref={ref}
-                                data-testid="selectRoleCircle"
-                                style={{'backgroundColor': color.color}}
-                                onClick={(): void => highlightCircle(ref, color)}
-                                onKeyDown={(e): void => handleKeyDownForHighlightCircle(e, ref, color)}
-                                className={`myTraitsCircle selectRoleCircle ${highlightDefaultCircle(color, index)} ${putBorderOnWhiteCircle(index)}`}/>
```
Comment (egriff38ford) [inline ui/src/Traits/EditTraitSection.tsx:212]: 👋  So happy to see this line go away

**Review by egriff38ford — APPROVED** [review]
🎉

**Review by sabdul14 — APPROVED** [review]

**Review by smle225 — APPROVED** [review]
- Change dropdown and text field focus state borderline color to #5463B0
- Remove shadow

**Comment by RandomAshes** [comment]
We're going to make a follow up PR for accessibility and to make those focus state changes.

**Pull request merged**